src/wazuh_modules/wm_sca.c

```diff
@@ -1001,6 +1029,23 @@ static int wm_sca_do_scan(cJSON *checks, OSStore *vars, wm_sca_t * data, int id,
         mdebug2("Initial rule-aggregator value por this type of rule is '%d'",  g_found);
         mdebug1("Beginning rules evaluation.");
 
+        w_expression_t * regex_engine = NULL;
```

Comment:
I propose a "collector/disposal" strategy to avoid tracking down every return, continue o loop break to avoid memory leakage.
With this approach, memory disposal is to be invoked which will reclaim all memory used.

Reply:
Removed proposal. Will apply a "by needed" basis approach